src/mlpack/methods/decision_tree/decision_tree_impl.hpp

```diff
@@ -624,10 +626,16 @@ void DecisionTree<FitnessFunction,
     }
 
     // Figure out counts of children.
-    arma::Row<size_t> childCounts(numClasses, arma::fill::zeros);
+    arma::Row<size_t> childCounts(numChildren, arma::fill::zeros);
     for (size_t i = begin; i < begin + count; ++i)
       childCounts[childAssignments[i - begin]]++;
 
+    //Initialize bestGain if recursive split is allowed
+    if(!NoRecursion)
```

Comment:
Just a few style comments---can you make sure that each comment is a fully punctuated sentence?  (i.e. period at the end).  Also we can put a space between `//` and `Initialize`, and a space between `if` and `(!NoRecursion)` to match the style guide.  These comments apply throughout, by the way, even though I've only pointed them out here.